apps/snitch_core/test/domain/splitters/weight_test.exs

```diff
@@ -0,0 +1,222 @@
+defmodule Snitch.Domain.Splitter.WeightTest do
+  use ExUnit.Case, async: true
+  use Snitch.DataCase
+
+  import Snitch.{OrderCase, StockCase}
+  import Snitch.Factory
+
+  alias Snitch.Core.Domain.Splitters.Weight, as: WeightSplitter
+  alias Snitch.Data.Schema.{StockItem, StockLocation, Order}
+  alias Snitch.Domain.Shipment
+
+  @sl_manifest %{
+    "default" => %{
+      default: true
+    },
+    "backup" => %{},
+    "origin" => %{}
+  }
+
+  @si_manifest %{
+    "default" => [
+      %{count_on_hand: 3, backorderable: true},
+      %{count_on_hand: 2, backorderable: true},
+      %{count_on_hand: 3, backorderable: true}
+    ],
+    "backup" => [
+      %{count_on_hand: 0},
+      %{count_on_hand: 0},
+      %{count_on_hand: 6}
+    ],
+    "origin" => [
+      %{count_on_hand: 3},
+      %{count_on_hand: 3},
+      %{count_on_hand: 3}
+    ]
+  }
+
+  setup :random_weight_variants
+  setup :stock_locations
+  setup :stock_items
+
+  describe "split/1" do
+    @tag weights: [10.0, 10.0, 10.0]
+    test "no package with weight over threshold", %{variants: vs} do
+      line_items = line_items_with_price(vs, [1, 1, 1])
+      order = %Order{id: 42, line_items: line_items}
+
+      splitted_packages =
+        order
+        |> Shipment.default_packages()
+        |> WeightSplitter.split()
+
+      # item = {delta quantity}
+      packages_to_assert = [
```

Comment:
This is not a change request, because I don't know of a better way myself :disappointed: 

These tests are a bit fragile because the "order" of packages is assumed. In the _very_ unlikely future, the order could change, and when that happens, these tests will have to be updated.
Helpers (reflection functions) like this [one](https://github.com/aviabird/snitch/blob/package-shipping-method-157490638/apps/snitch_core/test/domain/shipment_test.exs#L135) are not very useful here.